src/init/starship.ps1

```diff
@@ -211,8 +211,17 @@ $null = New-Module starship {
     )
 
     try {
-        Set-PSReadLineOption -ViModeIndicator script -ViModeChangeHandler {
-            [Microsoft.PowerShell.PSConsoleReadLine]::InvokePrompt()
+        # Combine user defined ViModeChangeHandler if it exists
+        if((Get-PSReadLineOption).ViModeChangeHandler){
+            Set-PSReadLineOption -ViModeIndicator script -ViModeChangeHandler $([scriptblock]::Create(
+                "[Microsoft.PowerShell.PSConsoleReadLine]::InvokePrompt()" +
+                "`n" +
+                (Get-PSReadLineOption).ViModeChangeHandler.ToString()
```

Comment:
Would it be possible to use and copy and execute the old handler without stringification?

Reply:
The ViModeChangeHandlers needs to be a script block. AFAIK there isn't a good way to to join scriptblocks without concatenating them this way. I'll ask around though just to be certain.

Do you have a specific concern? I suspect this is more of an edge case then not. And the resource cost would be minimal since it would only happen once at profile load. The stringification does not mean it's interpolated.

Reply:
This is going to create unexpected behavior, I recommend changing it to the following code

```pwsh
 Set-PSReadLineOption -ViModeIndicator script -ViModeChangeHandler $([scriptblock]::Create(
                [Microsoft.PowerShell.PSConsoleReadLine]::InvokePrompt() +
                "`n" +
                (Get-PSReadLineOption).ViModeChangeHandler.ToString()
```

Reply:
I was mostly concerned with strings causing not preserving the full semantics of any given handler and potentially issues with string concatenation itself.

@romero126 Do you still have concerns with the current approach?